platforms/nuttx/src/px4/microchip/mpfs/version/board_mcu_version.c

```diff
@@ -76,24 +98,38 @@ __EXPORT void board_get_uuid32(uuid_uint32_t uuid_words)
 
 int board_mcu_version(char *rev, const char **revstr, const char **errata)
 {
-	/* TODO: This is a stub for userspace build. Use some proper interface
-	 * to fetch the version from the kernel
-	 */
-	const char *str = "PolarFire MPFS250T";
-	*rev = '1';
-	*revstr = str;
-	*errata = 0;
-	return 0;
+	int ret = -1;
+
+	switch (hw_version) {
+
+		case MPFS_ICICLE:
+			*rev = '1';
```

Comment:
Just an idea: you can make a macro, which initializes a table, in board_config.h;

#define BOARD_REVISIONS \
{{"PolarFire MPFS250T"},'1',0}, \
{"PolarFire MPFS250T"},'1',0}}

Here you can define a table out of these:
const struct  {const char* revstr, char rev, int errata} hw_version_table[] = BOARD_REVISIONS;

And then you can just check the table length and index it with [hw_version] to get the revstr, errata etc. "hw_version_table[hw_version].revstr"

The idea is then that you can easily add new boards by just modifying board_config.h. Also, if there are other targets using this same code, they can make their own set of revisions by just defining the same table in their own board_config.h.